src/components/sketches/Scene3.ts

```diff
@@ -0,0 +1,136 @@
+    p.fill(PINK);
+    p.ellipse(BOUNDARY_X, BOUNDARY_Y, BOUNDARY_RADIUS);
+
+    // If too many particles on screen
+    if (particles.length > max) {
+      // Delete old particles (from beginning of array)
+      particles.splice(0, 10);
+    }
+
+    // Loop through the array and show each particle
+    for (let i = 0; i < particles.length; i++) {
+      particles[i].update();
+      particles[i].show();
+
+    }
+  };
+  /** Spawn ringParticles in ring with centre mouse x and y coords */
+  p.mouseClicked = () => {
+    particles.push(new Particle(p.mouseX, p.mouseY, p.random(0, 359), BALL_RADIUS, HOT_PINK));
+  };
+
+  class Particle {
+    public pos: p5.Vector;
+    private _vel: p5.Vector;
+    private _acc: p5.Vector;
+    private _radius: number;
+    public color:p5.Color;
+
+    constructor(x: number, y: number, theta: number, radius: number, color: p5.Color) {
+      this._radius = radius; // Radius
+      this.pos = p.createVector(x, y); // Position
+      this._vel = p.createVector(INIT_SPEED, INIT_SPEED); // Velocity
+      this._acc = p.createVector(ACCEL, ACCEL); // Acceleration
+      this.color = color;
+
+      this._vel.limit(INIT_SPEED + 2);
+      this._vel.rotate(theta); // Rotate velocity to the given rotation (avoids complicated maths)
+      this._acc.rotate(theta); // Rotate to the given rotation (avoids complicated maths)
+    }
+
+    public update () {
+      this.pos.add(this._vel); // Update position with velocity
+      this.edges(); // Check if hitting edge for rebound
+    }
+
+    public show() {
+      p.noStroke();
+      // Set colour based on average of x and y position
+      p.fill(this.color);
+      // Draw point
+      p.ellipse(this.pos.x, this.pos.y, this._radius);
+    }
+
+    public edges() {
+      const x_dist = this.pos.x - BOUNDARY_X;
+      const y_dist = this.pos.y - BOUNDARY_Y;
+
+      // If hitting edge, invert appropriate velocity
+      if (this.pos.x < 0 + this._radius) {
+        this._acc.x *= -1;
+        this._vel.x *= -1;
+        this.pos.x = 0 + this._radius;
+        this._vel.add(this._acc);
+      } else if (this.pos.x > p.width - this._radius) {
+        this._acc.x *= -1;
+        this._vel.x *= -1;
+        this.pos.x = p.width - this._radius;
+        this._vel.add(this._acc);
+      }
+      if (this.pos.y < 0 + this._radius) {
+        this._acc.y *= -1;
+        this._vel.y *= -1;
+        this.pos.y = 0 + this._radius;
+        this._vel.add(this._acc);
+      } else if (this.pos.y > p.height - this._radius) {
+        this._acc.y *= -1;
+        this._vel.y *= -1;
+        this.pos.y = p.height - this._radius;
+        this._vel.add(this._acc);
+      } else if (p.sqrt(x_dist * x_dist + y_dist * y_dist) < BOUNDARY_RADIUS / 2) {
+        this._vel.reflect(p.createVector(x_dist, y_dist));
+      } else {
+        if (p.sqrt(this._vel.x * this._vel.x + this._vel.y * this._vel.y) > INIT_SPEED) {
+          this._vel.limit(INIT_SPEED + 2);
+        }
+      }
```

Comment:
is this supposed to be in another if/else bracket? and if so pls label it to describe its behavior

Reply:
These series, represent whether the ball hits a wall or the circle, so its all one chain to ensure direction change of one wall only. I will add comments tho